From 77acb321c1a2d0a20be8650606fc63153ce5fa11 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 8 May 2022 11:45:54 +0200
Subject: [PATCH 0762/1010] evl/observable: fix misnomer EVL_CLONE_MASTER ->
 EVL_CLONE_UNICAST

Fixing a longstanding misnomer: this flag is really about sending
unicast notifications as opposed to broadcasting events to all
observers.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/uapi/evl/factory.h |  8 +++++++-
 kernel/evl/observable.c    | 12 ++++++------
 kernel/evl/thread.c        |  4 ++--
 3 files changed, 15 insertions(+), 9 deletions(-)

diff --git a/include/uapi/evl/factory.h b/include/uapi/evl/factory.h
index 50dc36bdc6..fde46bc92b 100644
--- a/include/uapi/evl/factory.h
+++ b/include/uapi/evl/factory.h
@@ -22,11 +22,17 @@ struct evl_element_ids {
 #define EVL_CLONE_PRIVATE	(0 << 16)
 #define EVL_CLONE_OBSERVABLE	(1 << 17)
 #define EVL_CLONE_NONBLOCK	(1 << 18)
-#define EVL_CLONE_MASTER	(1 << 19)
+#define EVL_CLONE_UNICAST	(1 << 19)
 #define EVL_CLONE_INPUT		(1 << 20)
 #define EVL_CLONE_OUTPUT	(1 << 21)
 #define EVL_CLONE_COREDEV	(1 << 31)
 #define EVL_CLONE_MASK		(((__u32)-1 << 16) & ~EVL_CLONE_COREDEV)
+/*
+ * Deprecated: this is a longstanding misnomer. This flag is really
+ * about sending unicast notifications as opposed to broadcasting
+ * events to all observers.
+ */
+#define EVL_CLONE_MASTER	EVL_CLONE_UNICAST
 
 struct evl_clone_req {
 	__u64 name_ptr;		/* (const char __user *name) */
diff --git a/kernel/evl/observable.c b/kernel/evl/observable.c
index 65cfa4ba9e..dab0272ccd 100644
--- a/kernel/evl/observable.c
+++ b/kernel/evl/observable.c
@@ -20,7 +20,7 @@
 #define EVL_NOTIFY_INITIAL	(1 << 31)
 
 #define EVL_OBSERVABLE_CLONE_FLAGS	\
-	(EVL_CLONE_PUBLIC|EVL_CLONE_OBSERVABLE|EVL_CLONE_MASTER)
+	(EVL_CLONE_PUBLIC|EVL_CLONE_OBSERVABLE|EVL_CLONE_UNICAST)
 
 /*
  * We want any kind of threads to be able to subscribe to an
@@ -184,7 +184,7 @@ static int add_subscription(struct evl_observable *observable,
 	 * the observer and writability tracking to be atomically
 	 * updated, so nested locking is required. Observers are
 	 * linked to the observable's list by order of subscription
-	 * (FIFO) in case users make such assumption for master mode
+	 * (FIFO) in case users make such assumption for unicast mode
 	 * (which undergoes round-robin).
 	 */
 	raw_spin_lock_irqsave(&observable->lock, flags);
@@ -598,9 +598,9 @@ static bool notify_all(struct evl_observable *observable,
 	return do_flush;
 }
 
-static bool is_pool_master(struct evl_observable *observable)
+static bool is_pool_unicast(struct evl_observable *observable)
 {
-	return !!(observable->element.clone_flags & EVL_CLONE_MASTER);
+	return !!(observable->element.clone_flags & EVL_CLONE_UNICAST);
 }
 
 static bool push_notification(struct evl_observable *observable,
@@ -622,10 +622,10 @@ static bool push_notification(struct evl_observable *observable,
 	nfr.date = date;
 
 	/*
-	 * Feed one observer/worker if master, or broadcast to all
+	 * Feed one observer/worker if unicast, or broadcast to all
 	 * observers.
 	 */
-	if (is_pool_master(observable))
+	if (is_pool_unicast(observable))
 		do_flush = notify_single(observable, &nfr, &len);
 	else
 		do_flush = notify_all(observable, &nfr, &len);
diff --git a/kernel/evl/thread.c b/kernel/evl/thread.c
index 215d453b05..9670b37402 100644
--- a/kernel/evl/thread.c
+++ b/kernel/evl/thread.c
@@ -43,7 +43,7 @@
 #include <trace/events/evl.h>
 
 #define EVL_THREAD_CLONE_FLAGS	\
-	(EVL_CLONE_PUBLIC|EVL_CLONE_OBSERVABLE|EVL_CLONE_MASTER)
+	(EVL_CLONE_PUBLIC|EVL_CLONE_OBSERVABLE|EVL_CLONE_UNICAST)
 
 int evl_nrthreads;
 
@@ -2436,7 +2436,7 @@ thread_factory_build(struct evl_factory *fac, const char __user *u_name,
 		if (ret)
 			goto fail_observable_dev;
 		observable = observable;
-	} else if (clone_flags & EVL_CLONE_MASTER) {
+	} else if (clone_flags & EVL_CLONE_UNICAST) {
 		ret = -EINVAL;
 		goto fail_observable;
 	}
-- 
2.30.2

